internal/pipeline/step_command.go

```diff
@@ -89,25 +89,62 @@ func (c *CommandStep) unmarshalMap(m *ordered.MapSA) error {
 }
 
 // SignedFields returns the default fields for signing.
-func (c *CommandStep) SignedFields() map[string]string {
+func (c *CommandStep) SignedFields() (map[string]string, error) {
+	plugins := ""
+	if len(c.Plugins) > 0 {
+		// TODO: Reconsider using JSON here - is it stable enough?
```

Comment:
Ordered Maps are an issue, yeah? I think we can overcome it if we marshal all the unordered stuff into ordered equivalents.

Another one could be floating point representation, and precision. If the backend or database changes the precision of a number, we could have an issue.

Reply:
> Ordered Maps are an issue, yeah? I think we can overcome it if we marshal all the unordered stuff into ordered equivalents.

My thinking on this is: the pipeline parser ultimately shoves whatever `ordered.DecodeYAML` produces for the plugin config into `Plugin.Config`, which will tend to be either `*ordered.Map[string, any]` or `nil`, preserving input order. As long as the backend does it the same, it should be equal.

> Another one could be floating point representation, and precision. If the backend or database changes the precision of a number, we could have an issue.

This kind of thing _does_ concern me.